src/node/rpc/frontend.h

```diff
@@ -273,6 +273,10 @@ namespace ccf
       {
         if (attempts > 0)
```

Comment:
The retry loop should be moved higher up in this function, so that we lookup the endpoint (useful for JS apps) and authenticate the caller based on a new KV `read_version`. I believe this is safe but haven't included the change in this PR. In particular:
- If the request was forwarded to the primary, the transaction will not commit so no retry there.
- If the request was forwarded to us (primary), the transaction may conflict but that's fine too.

Am I missing anything?

Reply:
That sounds right, but I seem to remember there were some problems with that. One thing that comes to mind was discovering from re-reading the endpoint that it shouldn't forward (anymore), while executing it in a forwarded context. Let's discuss this with @eddyashton, I agree it's best done separately.